processor/src/main/java/org/mapstruct/ap/internal/model/BeanMappingMethod.java

```diff
@@ -1520,7 +1520,9 @@ private ReportingPolicyGem getUnmappedSourcePolicy() {
         private void reportErrorForUnmappedSourcePropertiesIfRequired() {
             ReportingPolicyGem unmappedSourcePolicy = getUnmappedSourcePolicy();
 
-            if ( !unprocessedSourceProperties.isEmpty() && unmappedSourcePolicy.requiresReport() ) {
+            // If we have ignoreByDefault = true, unprocessed source properties are not an issue.
+            if ( !method.getOptions().getBeanMapping().isignoreByDefault() && !unprocessedSourceProperties.isEmpty()
+                && unmappedSourcePolicy.requiresReport()) {
```

Comment:
Let's move this logic in `getUnmappedSourcePolicy()`.  We already have something similar there for forged methods

Reply:
Ok, will do!